File: quickpbt_fs/Teaser.fs
namespace quickpbt

open Xunit
open FsCheck
open FsCheck.Xunit

/// Contrasts a unit test with a property test
module Teaser =
  let daysInAWeek   =  7
  let hoursInAWeek  = 24 * daysInAWeek

  [<Fact>]
  let ``days should equal hours`` () =
    let today = Date.Now //NOTE: single, hard-coded value

    let days  = today + Time.FromDays(daysInAWeek)
    let hours = today + Time.FromHours(hoursInAWeek)

    Assert.Equal(days, hours)

  [<Property>]
  let ``unit of time should not effect addition`` (today :Date) =
    //NOTE: lots of different, random values
    let days  = today + Time.FromDays(daysInAWeek)
    let hours = today + Time.FromHours(hoursInAWeek)

    days = hours

File: quickpbt_cs/Teaser.cs
using Xunit;
using FsCheck;
using FsCheck.Xunit;

namespace quickpbt
{
  /* domain under test */
  using Date = System.DateTimeOffset;
  using Time = System.TimeSpan;
  using Zone = System.TimeZoneInfo;

  /// <summary>
  /// Contrasts a unit test with a property test
  /// </summary>
  public sealed class Teaser
  {
    static readonly int DaysInAWeek   =  7;
    static readonly int HoursInAWeek  = 24 * DaysInAWeek;

    [Fact]
    public void days_should_equal_hours()
    {
      var today = Date.Now; //NOTE: single, hard-coded value

      var days  = today + Time.FromDays(DaysInAWeek);
      var hours = today + Time.FromHours(HoursInAWeek);

      Assert.Equal(days, hours);
    }

    [Property]
    public bool unit_of_time_should_not_effect_addition(Date anyDate)
    {
      //NOTE: lots of different, random values
      var days  = anyDate + Time.FromDays(DaysInAWeek);
      var hours = anyDate + Time.FromHours(HoursInAWeek);

      return days == hours;
    }
  }
}
